### DISCUSSION TITLE
v0.8.4

### DISCUSSION THREAD

**Discussion by justinxzhao** [ludwig-ai/ludwig#3639] [General]
## What's Changed
* Add codellama to tokenizer list for set_pad_token by @Infernaught in https://github.com/ludwig-ai/ludwig/pull/3598
* Set default eval batch size to 2 for LLM fine-tuning by @arnavgarg1 in https://github.com/ludwig-ai/ludwig/pull/3599
* [CI] Explicitly set eval batch size in determinism tests, introduce a new integration test group, and exclude slow tests. by @justinxzhao in https://github.com/ludwig-ai/ludwig/pull/3590
* [CI] Run sudo apt-get update in GHAs. by @justinxzhao in https://github.com/ludwig-ai/ludwig/pull/3608
* Store steps_per_epoch in Trainer by @hungcs in https://github.com/ludwig-ai/ludwig/pull/3601
* Updated characters, underscore and comma preprocessors to be TorchScriptable. by @martindavis in https://github.com/ludwig-ai/ludwig/pull/3602
* [CI] Deflake: Explicitly set eval batch size for mlflow test. by @justinxzhao in https://github.com/ludwig-ai/ludwig/pull/3612
* Fix registration for char error rate. by @justinxzhao in https://github.com/ludwig-ai/ludwig/pull/3604
* fix: Load 8-bit quantized models for eval after fine-tuning by @jeffkinnison in https://github.com/ludwig-ai/ludwig/pull/3606
* Add Code Alpaca and Consumer Complaints Datasets by @connor-mccorm in https://github.com/ludwig-ai/ludwig/pull/3611
* Add support for gradient checkpointing for LLM fine-tuning by @arnavgarg1 in https://github.com/ludwig-ai/ludwig/pull/3613
* Bump min support transformers to 4.33.0 by @tgaddair in https://github.com/ludwig-ai/ludwig/pull/3616
* [CI] Fix failing tests on master by @arnavgarg1 in https://github.com/ludwig-ai/ludwig/pull/3617
* Eliminate short-circuiting for loading from local by @Infernaught in https://github.com/ludwig-ai/ludwig/pull/3600
* Refactor integration tests into matrix by @tgaddair in https://github.com/ludwig-ai/ludwig/pull/3618
* fix: Check underlying model device type when moving 8-bit quantized models to GPU at eval by @jeffkinnison in https://github.com/ludwig-ai/ludwig/pull/3622
* Fixed range validation for text generation penalty parameters by @tgaddair in https://github.com/ludwig-ai/ludwig/pull/3623
* Update comment for predict to update Ludwig docs by @Infernaught in https://github.com/ludwig-ai/ludwig/pull/3535
* Avoid deprecation warnings on pandas Series.fillna by @carlogrisetti in https://github.com/ludwig-ai/ludwig/pull/3631
* QoL: Default to using fast tokenizer for Llama models by @arnavgarg1 in https://github.com/ludwig-ai/ludwig/pull/3625
* fixed typo in EfficientNet's model variant from v2_ to v2_s by @saad-palapa in https://github.com/ludwig-ai/ludwig/pull/3628
* Add pytorch profiler and additional tensorboard logs for GPU memory usage. by @justinxzhao in https://github.com/ludwig-ai/ludwig/pull/3607
* Pin minimum transformers version to `4.33.2` by @arnavgarg1 in https://github.com/ludwig-ai/ludwig/pull/3637

## New Contributors
* @saad-palapa made their first contribution in https://github.com/ludwig-ai/ludwig/pull/3628

**Full Changelog**: https://github.com/ludwig-ai/ludwig/compare/v0.8.3...v0.8.4

<hr /><em>This discussion was created from the release <a href='https://github.com/ludwig-ai/ludwig/releases/tag/v0.8.4'>v0.8.4</a>.</em>
